ldms/src/ldmsd/ldmsd_stream.c

```diff
@@ -775,7 +777,9 @@ int ldmsd_stream_publish_file(const char *stream, const char *type,
 close_xprt:
 	ldms_xprt_close(x);
 err:
+	free(b);
 	free(buffer);
+	buffer = NULL;
```

Comment:
So this is only here if we assume that the buffer pointer is static, which we have already decided it should not be.